crates/jstzd/src/task/octez_node.rs

```diff
@@ -174,7 +176,22 @@ impl Task for OctezNode {
 
     /// Conducts a health check on the running task.
     async fn health_check(&self) -> Result<bool> {
-        todo!()
+        // Returns whether or not the node is ready to answer to requests.
+        // https://gitlab.com/tezos/tezos/-/raw/2e84c439c25c4d9b363127a6685868e223877034/docs/api/rpc-openapi.json
+        let res =
+            reqwest::get(format!("http://{}/health/ready", &self.config.rpc_endpoint))
+                .await;
+        if res.is_err() {
+            return Ok(false);
+        }
+        let body = res
+            .unwrap()
```

Comment:
perhaps beter to return error with `?`

Reply:
Here I don't think we should return the error. The cases I imagined here are like before the RPC server is even launched. These requests would then fail with something like connection refused errors, and these should be considered as "the node not being ready yet" rather than severe errors. The `json::` case, on the other hand, is where errors should be returned since if there is any error in de-serialisation, it might indicate that the implementation here is outdated, e.g. we did not catch the changes in RPC, and we would need to fix that.